app/lib/event_store/event.rb

```diff
@@ -64,6 +66,36 @@ def load(str)
 
     scope :stale, -> { where.has { created_at <= TTL.ago } }
 
+    class EventRollbackError < StandardError
+      include Bugsnag::MetaData
+
+      def initialize(error:, event:)
+        event_details = event.to_h.as_json
+        self.bugsnag_meta_data = { event: event_details }
+
+        super "Error raised trying to roll back #{event}: #{self}, event details: #{event_details}"
+        set_backtrace error.backtrace
+      end
+    end
+
+    # Since Rails 6.1 the EventStore::Event records (among others) are added to the
+    # list of the objects, belonging to a current transaction, and #rolledback! is called on them
+    # if the transaction is rolled back on the DB.
+    # The problem is that this triggers the Event's `load` and `deserialize` (to restore the original state), and if
+    # the object referenced in the event via GlobalID does not exist, the deserialization of the event will fail.
+    # This causes some tests (that have explicit transactions and rollbacks) to fail, and can potentially cause issues
+    # in production as well. Currently, it's safe to ignore these errors for Events rollback. But in case any transactional
+    # callbacks are added to Events in future, we will report this to BugSnag.
```

Comment:
```suggestion
    # Since Rails 6.1 the EventStore::Event records (among others) are added to the
    # list of the objects, belonging to a current transaction, and #rolledback! is called on them
    # if the transaction is rolled back on the DB.
    # see https://github.com/rails/rails/commit/77f7b2df#diff-8dd03b7fb9b72a3bd338955c1de75652d60453230c6544f3851c0d0b3746a675L345-R349
    # The problem is that this triggers the Event's `load` and `deserialize` (to restore the original state), and if
    # the object referenced in the event via GlobalID does not exist, the deserialization of the event will fail.
    # Prevent an exception during rollback but try to report when that's unsafe.
```